core/include/usBundle.h

```diff
@@ -47,143 +48,297 @@ typedef ServiceReference<void> ServiceReferenceU;
 /**
  * \ingroup MicroServices
  *
- * Represents a CppMicroServices bundle.
+ * An installed bundle in the Framework.
  *
- * <p>
- * A <code>%Bundle</code> object is the access point to a CppMicroServices bundle.
- * Each CppMicroServices bundle has an associated <code>%Bundle</code> object.
+ * A <code>%Bundle</code> object is the access point to define the lifecycle of an
+ * installed bundle. Each bundle installed in the CppMicroServices environment has
+ * an an associated <code>%Bundle</code> object.
  *
- * <p>
- * A bundle has unique identity, a <code>long</code>, chosen by the
- * framework. This identity does not change during the lifecycle of a bundle.
+ * A bundle has a unique identity, a <code>long</code>, chosen by the
+ * Framework. This identity does not change during the lifecycle of a bundle.
+ * Uninstalling and then reinstalling the bundle creates a new unique identity.
  *
- * <p>
- * A bundle can be in one of four states:
- * <ul>
- * <li>INSTALLED
- * <li>STARTED
- * <li>STOPPED
- * <li>UNINSTALLED
- * </ul>
- * <p>
- * You can determine whether a bundle is started or not by using IsStarted().
+ * A bundle can be in one of six states:
+ * - STATE_UNINSTALLED
+ * - STATE_INSTALLED
+ * - STATE_RESOLVED
+ * - STATE_STARTING
+ * - STATE_STOPPING
+ * - STATE_ACTIVE
  *
- * <p>
- * A bundle can only execute code when its state is <code>STARTED</code>.
- * A <code>STOPPED</code> bundle is a
- * zombie and can only be reached because it was started before. However,
- * stopped bundles can be started again.
+ * Values assigned to these states have no specified ordering; they represent
+ * bit values that may be ORed together to determine if a bundle is in one of
+ * the valid states.
+ *
+ * A bundle should only have active threads of execution when its state is one
+ * of {@code STATE_STARTING},{@code STATE_ACTIVE}, or {@code STATE_STOPPING}.
+ * A {@code STATE_UNINSTALLED} bundle can not be set to another state; it is a
+ * zombie and can only be reached because references are kept somewhere.
  *
- * <p>
  * The framework is the only entity that is allowed to create
- * <code>%Bundle</code> objects.
+ * <code>%Bundle</code> objects, and these objects are only valid within the
+ * Framework that created them.
+ *
+ * Bundles have a natural ordering such that if two {@code Bundle}s have the
+ * same {@link #GetBundleId() bundle id} they are equal. A {@code Bundle} is
+ * less than another {@code Bundle} if it has a lower {@link #GetBundleId()
+ * bundle id} and is greater if it has a higher bundle id.
+ *
  *
  * @remarks This class is thread safe.
  */
-class US_Core_EXPORT Bundle : public std::enable_shared_from_this<Bundle>
+class US_Core_EXPORT Bundle
 {
 
 public:
 
-  Bundle(const Bundle &) = delete;
-  Bundle& operator=(const Bundle&) = delete;
+  typedef Clock::time_point TimeStamp;
+
+  enum State : uint32_t {
```

Comment:
You're right. I had forgotten about the macro define in wingdi.h.
I had initially thought clients could deal with it on their own (e.g. by using ```#undef ERROR```) however that puts a burden on clients.

Your approach to make the identifiers less generic looks to be the best way to avoid this macro name collision problem.